internal/charmstore/migrations.go

```diff
@@ -175,12 +175,12 @@ func addDevelopmentACLs(db StoreDatabase) error {
 	baseEntities := db.BaseEntities()
 	var baseEntity mongodoc.BaseEntity
 	iter := baseEntities.Find(bson.D{{
-		"developmentacls", bson.D{{"$exists", false}},
-	}}).Select(bson.D{{"_id", 1}, {"acls", 1}}).Iter()
+		"channelacls.development", bson.D{{"$exists", false}},
+	}}).Select(bson.D{{"_id", 1}, {"channelacls", 1}}).Iter()
 	defer iter.Close()
 	for iter.Next(&baseEntity) {
 		if err := baseEntities.UpdateId(baseEntity.URL, bson.D{{
-			"$set", bson.D{{"developmentacls", baseEntity.ACLs}},
+			"$set", bson.D{{"channelacls.development", baseEntity.ChannelACLs[mongodoc.DevelopmentChannel]}},
```

Comment:
I think this should populate "channelacls.development" with the value from `baseEntity.ChannelACLs[mongodoc.UnpublishedChannel]`.